xio-core/src/main/java/com/xjeffrose/xio/http/ClientChannelInitializer.java

```diff
@@ -0,0 +1,70 @@
+package com.xjeffrose.xio.http;
+
+import com.xjeffrose.xio.core.XioIdleDisconnectHandler;
+import com.xjeffrose.xio.core.XioMessageLogger;
+import com.xjeffrose.xio.pipeline.Pipelines;
+import com.xjeffrose.xio.tracing.XioTracing;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelInitializer;
+import java.util.function.Supplier;
+import lombok.val;
+
+public class ClientChannelInitializer {
+  public static final String sslHandlerName = "ssl handler";
+  public static final String negotiationHandlerName = "negotiation handler";
+  public static final String codecName = "codec";
+  public static final String tracingName = "distributed tracing";
+  public static final String applicationCodecName = "application codec";
+  public static final String idleHandlerName = "idle handler";
+  public static final String messageLoggingName = "message logging";
+  public static final String requestBufferName = "request buffer";
+  public static final String appHandlerName = "app handler";
+
+  private final ClientState state;
+  private final Supplier<ChannelHandler> appHandler;
+  private final XioTracing tracing;
+
+  public ClientChannelInitializer(
+      ClientState state, Supplier<ChannelHandler> appHandler, XioTracing tracing) {
+    this.state = state;
+    this.appHandler = appHandler;
+    this.tracing = tracing;
+  }
+
+  public ChannelInitializer createChannelInitializer() {
```

Comment:
unless it's making testing easier, I'm not seeing a good reason for composing with `ChannelInitializer` instead of inheiriting it. Right now `ClientChannelInitializer` is a `ChannelInitializer` Factory and not a `ChannelInitializer`.

Reply:
ok I can buy that, adjusting